### ISSUE TITLE
Fails to infer template parameter type in std::vector<T> when T is a pointer type

### ISSUE THREAD

**Issue by jslee02** [personalrobotics/chimera#216]
Here is a sample code:
```cpp
#include <vector>

namespace chimera_test {
namespace common {

template <typename T>
inline void vector_param(T val) { }

template <typename T>
inline void vector_param(const std::vector<T>& vec) { }

class Node
{
public:
  Node() = default;
};

class Skeleton
{
public:
  Skeleton() = default;
  inline void foo()
  { 
    auto vec = std::vector<Node>();
    auto ptrVec = std::vector<Node*>();
    
    common::vector_param(vec);  // ok

    // error: ‘Node’ was not declared in this scope
    // sm.def(
    //     "vector_param",
    //     +[](const std::vector<Node *> & vec) -> void 
    //         { return chimera_test::common::vector_param<chimera_test::common::Node *>(vec); },
    //     ::pybind11::arg("vec"));
    common::vector_param(ptrVec);

    common::vector_param(Node());      // ok
    common::vector_param(new Node());  // ok
  }
};

} // namespace dynamics
} // namespace chimera_test
```

Note that the scope of `Node` in `const std::vector<Node *>&` isn't parsed.

This happens for both of Boost.Python and pybind11 where Boost.Python additionally complains about the held type.

**Comment by jslee02** [comment]
Update:
Investigating further, [this line](https://github.com/personalrobotics/chimera/blob/master/external/cling/src/cling_utils_AST.cpp#L1371) transforms:
```cpp
vector<chimera_test::nested_namespace::Node>
vector<chimera_test::nested_namespace::Node *>
```
to
```cpp
std::vector<chimera_test::nested_namespace::Node>
std::vector<Node *>  // wrong !
```

**Comment by jslee02** [comment]
Resolved by #220
